sunpy/time/time.py

```diff
@@ -133,26 +134,115 @@ def _parse_dt64(dt):
     return datetime(*(dt.astype(datetime).timetuple()[:6]))
 
 
+@singledispatch
+def convert_time(time_string, **kwargs):
+    # default case when no type matches
+    raise ValueError("'{tstr!s}' is not a valid time string!".format(tstr=time_string))
+
+
+@convert_time.register(pandas.Timestamp)
+def convert_time_pandasTimestamp(time_string, **kwargs):
+    return time_string.to_pydatetime()
+
+
+@convert_time.register(pandas.Series)
+def convert_time_pandasSeries(time_string, **kwargs):
+    if 'datetime64' in str(time_string.dtype):
+        return np.array([dt.to_pydatetime() for dt in time_string])
+    else:
+        convert_time.dispatch(object)(time_string, **kwargs)
+
+
+@convert_time.register(pandas.DatetimeIndex)
+def convert_time_pandasDatetimeIndex(time_string, **kwargs):
+    return time_string._mpl_repr()
+
+
+@convert_time.register(datetime)
+def convert_time_datetime(time_string, **kwargs):
+    return time_string
+
+
+@convert_time.register(date)
+def convert_time_date(time_string, **kwargs):
+    return datetime.combine(time_string, time())
+
+
+@convert_time.register(tuple)
+def convert_time_tuple(time_string, **kwargs):
+    return datetime(*time_string)
+
+
+@convert_time.register(float)
+@convert_time.register(int)
+def convert_time_float(time_string, **kwargs):
```

Comment:
What about quantities, like 10 * u.second ?

Reply:
I am semi-relucant to make parse_time assume even more things are utime. Given that this PR is only phase 1 in a large redactor (and will not be in 0.9) I propose we leave this as-is for now.